### PR TITLE
Support IDN and punicode TLDs

### LINKED ISSUE #200: URL matcher doesn't recognize new national TLDs like .рф
Autolinker doesn't recognize valid urls like http://россия.рф or their punicode equivalent http://xn--d1acufc.xn--p1ai.

(GitHub doesn't get Cyrillic one either 😐)

It seems like a list of common TLDs is used in https://github.com/gregjacobs/Autolinker.js/blob/1.4.1/src/RegexLib.js#L57

I'd suggest either using a permissive alphanumeric unicode regex like /[-\w]{2,63}/u or importing the [official list](http://data.iana.org/TLD/tlds-alpha-by-domain.txt) in its entirety, and also providing unicode equivalents to punicode TLDs like xn--...

### PR DESCRIPTION
**Pull request by tsx**
Fixes #200

### CODE REVIEW

File: tests/AutolinkerSpec.js [hunk 1]
```diff
@@ -305,15 +308,6 @@ describe( "Autolinker", function() {
 				} );
 
 
-				it( 'should not match local urls with numbers when NOT prefixed with http://', function() {
```
Comment (olafleur) [inline tests/AutolinkerSpec.js]: Why did you remove that test ?

File: tests/matcher/UrlSpec.js [hunk 2]
```diff
@@ -106,13 +106,6 @@ describe( "Autolinker.matcher.Url", function() {
 		});
 
 
-		it( 'should not match a local URL with numbers that does not have the http:// before', function() {
```
Comment (olafleur) [inline tests/matcher/UrlSpec.js]: Same question here

**Comment by gregjacobs** [comment]
Hey @tsx. So there was a reason why we originally included a tld's list rather than fuzzy matching for a tld. We basically wanted to remove false positive matches as much as possible.

I think what we should do is actually instead is remove the need to maintain the list of tld's by hand, and make it part of the build process to pull the latest list from the Internet instead.

Thoughts @olafleur?

**Comment by olafleur** [comment]
I agree with @gregjacobs . Sorry @tsx , I was the one to mislead you.

That would be pretty awesome to have an up-to-date of TLDs at build time (or release time?).

**Comment by tsx** [comment]
@gregjacobs @olafleur I see your point. I have updated this PR to include an updated TLD list as well as a `gulp update-tld-list` to download and create a new regex from official IANA data. Removed (previously) tests are back and passing now.

I have not plugged `update-tld-list` into the main build target to avoid hitting IANA servers on every build.

**Comment by olafleur** [comment]
This is awesome @tsx !

I will let @gregjacobs take a look but LGTM for me !

**Comment by olafleur** [comment]
Thanks for your contribution !

**Comment by olafleur** [comment]
Published in version 1.4.2

**Comment by gregjacobs** [comment]
@tsx Sorry for not commenting on the updated version of this PR sooner. Just wanted to say, awesome, awesome, epic job on this! I'm very impressed! 

Thanks so much for doing this. This is a wonderful contribution to Autolinker!
